### DISCUSSION TITLE
WOCHE 03 - UCRS und Ausbau von SRS

### DISCUSSION THREAD

**Discussion by mausio** [mausio/corgi-shop-doc#5] [Blogs]
Moin,
Diese Woche haben wir das SRS um drei Aktivitätsdiagramme ergänzt, die Sequenzdiagramme ergänzt, am Front und Back End weiterentwickelt und zwei UCRS erstellt .

1. [Aktivitätsdiagramme](https://github.com/mausio/corgi-shop-doc/blob/main/srs/srs.md): Die Aktivitätsdiagramme repräsentieren die Abfolge von Aktivitäten, wenn der*die User eine Aktion durchführt wie z.B. einen Login.
2. [UCRS](https://github.com/mausio/corgi-shop-doc/blob/main/ucrs): Die UCRS zeigen die unterteilten Funktionen oder Dateien bzw. Navigationsoptionen und wie diese verknüpft sind. Dabei wird eine Beschreibung, Funktionsweise und Abfolge erklärt, wie was wo macht. 
3. Sequenzdiagramme: Jetzt getrennt von einander und ergänz
4. Front End: Corgis werden nun gefetched, in einer Liste dargestellt und enthalten Details, wenn man auf sie clicked. Der Code wurde nun auch cleaner gestaltet und boilerplate, so wie dead code wurde entfernt.
5. Back End: Corgis sind nun in einer Datenbank anlegbar und werden per API-Call gefetched über localhost:8080/api/corgis
6. Mael ist in der Entwicklung vom Merch- Back End beschäftigt, wobei es darum geht zwei Endpunkte an der API zu verfügung zu stellen, die die Kategorien mit Namen und Beispielbild in einer Liste e.g. darstellt und beim Click die Produkte aufführt; Das Front End wird daraufhin entwickelt.

**Comment by JannisFuchs** [comment]
Das sieht schon mal gut aus. Die Diagramme sind verständlich und die UI passt auch.

**Comment by Fefke** [comment]
Hey, euer Arbeitsstand sieht stabil aus 💪.
Bei eurem Tech-Stack ist mir ein Formatierungs-Fehler aufgefallen 
_(so ein paar Newlines kann man da noch ergänzen)_

Des Weiteren kann ich nicht ganz entschlüsseln zu welchem Use-Case die Aktivitäts-/ und Sequenzdiagramme dazugehören,
vielleicht könnt ihr das für nächste Woche noch ein wenig aufschlüsseln.

Die Sequenzdiagramme sind schlank, dafür soweit richtig und straight forward,
schön alle Parteien aufgeschlüsselt.

Habt ihr eure Mockups schon in X-Code erstellt? Sieht gut aus ^^.

**Comment by kattyterra** [comment]
Insgesamt sieht ganz gut aus, vielleicht könnte man die Sequnzdiagramme etwas aufteilen, damit sie etwas übersichtlicher werden

**Comment by Phylyp1** [comment]
Klasse Arbeitsstand! Die Diagramme sind alle sehr Verständlich

**Comment by mausio** [comment]
Wow.... I mean, wow! Das habt ihr echt gut gemacht!
